generative_ai/embedding_model_tuning.py

```diff
@@ -34,7 +34,7 @@ def tune_embedding_model(
     batch_size: int = 50,
     iterations: int = 300,
 ) -> pipeline_jobs.PipelineJob:
-    match = re.search(r"(.+)(-autopush|-staging)?-aiplatform.+", api_endpoint)
+    match = re.search(r"([\w]+-[\w]+).*-aiplatform.+", api_endpoint)
```

Comment:
Wouldn't this be easier with a string split instead of regex search?

Reply:
I think re.search is cleaner than two splits and a join:

* api_endpoint = "us-central1-xxx-yyy.aiplatform.googleapis.com"
* location1 = re.search(r"^([\w]+-[\w]+)", api_endpoint)[1]
* location2 = "-".join(api_endpoint.split(".")[0].split("-")[:2])  # doesn't look simpler to me.

Reply:
It doesn't work for this case:
* api_endpoint = "us-central1-xxx-yyy.zzzz.aiplatform.googleapis.com"

You have to do two splits and a join to handle other funny cases.

Reply:
Ah. That's icky. Got it.